docs/scripts/examples/simple_mlj_demo/simple_mlj_demo.jl

```diff
@@ -0,0 +1,321 @@
+# Evaluating over 2 metamodels: 100%[=========================] Time: 0:00:23
+# Evaluating over 2 metamodels: 100%[=========================] Time: 0:00:03
+# Evaluating over 2 metamodels: 100%[=========================] Time: 0:00:16
+# Evaluating over 2 metamodels: 100%[=========================] Time: 0:00:03
+# Evaluating over 2 metamodels: 100%[=========================] Time: 0:00:23
+# Evaluating over 2 metamodels: 100%[=========================] Time: 0:00:08
+# Machine{DeterministicTunedModel{MLJTreeParzenTuning,…}} @ 4…10
+
+# It's worth noting, that this changed behaviour of optimiser but didn't
+# introduce parallelism directly: That's up to MLJ. So lets parallelise.
+# It wouldn't really accelerate computations in this case possibly due to
+# overheads, but is an option for larger tasks. Also note that we can instead
+# parallelise via `resampling_acceleration` if we don't want to alter behaviour of
+# optimiser, but this might be more limited.
+
+suggestion = Dict(key => getproperty(best_model, key) for key in keys(space))
+search = TreeParzen.MLJTreeParzen.MLJTreeParzenSpace(space, suggestion)
+
+tuning = MLJTuning.TunedModel(
+    model=model_tpl,
+    ranges=space,
+    tuning=TreeParzen.MLJTreeParzen.MLJTreeParzenTuning(;random_trials=3, max_simultaneous_draws=2, linear_forgetting=50),
+    n=NUM_TP_ITER_SMALL,
+    resampling=MLJ.CV(nfolds=NUM_CV_FOLDS),
+    measure=MLJ.mav,
+    acceleration=ComputationalResources.CPUProcesses(),
+)
+
+mach = MLJ.machine(tuning, train_features, log.(train_targets))
+MLJ.fit!(mach)
+
+
+# An interesting feature of note is that TreeParzen supports "tree-structured"
+# parameter spaces; hence the name. Originally conceived for optimising
+# Deep Belief Networks (DBNs), where parameters take form such as number of layers,
+# and how many nodes are within each layer. Here, we can see the idea that if
+# we have 2 layers, then number of nodes in layer 3 as a parameter is not relevant.
+# We can construct a similar example using XGBoost - either we can use boosted trees
+# or boosted linears. Some parameters are relevant in both cases, but even so we
+# would want to model them differently, e.g. `num_iterations` might need to be lower for
+# boosting trees than for boosting linear functions.
+
+# First, create a container model (because of space nesting, we need to do this)
+# plus its constructor:
+
+mutable struct tuned_xgb <: MLJ.Deterministic
+    xgb::XGBoostRegressor
+end
+tuned_xgb(;xgb=Dict{Symbol, Any}()) = tuned_xgb(XGBoostRegressor(;xgb...))
+
+# quick fit and predict methods
+MLJ.fit(t::tuned_xgb, verbosity::Int, X, y, w=nothing) = MLJ.fit(t.xgb, verbosity, X, y, w)
+MLJ.predict(t::tuned_xgb, fitted, X) = MLJ.predict(t.xgb, fitted, X)
+
+# Define search parameters for the tree search space
+tree_space = Dict(
+    :booster => "gbtree",
+    :num_round => TreeParzen.HP.QuantUniform(:num_round_tree, 50., 750., 1.),
```

Comment:
Perhaps a `using TreeParzen` at the top so you can do `:num_round => HP.QuantUniform(blah...`?

Reply:
hmm I actually like imports because it's easier to just see where the function came from...it doesn't make it shorter, though.